src/engine/platform/ym2203ext.cpp

```diff
@@ -738,12 +779,20 @@ int DivPlatformYM2203Ext::init(DivEngine* parent, int channels, int sugRate, con
   extSys=true;
 
   reset();
-  return 9;
+  return psgChanOffs+2+4; // 3xPSG + optional CSM + 2xFM + 4xOP
 }
 
 void DivPlatformYM2203Ext::quit() {
   DivPlatformYM2203::quit();
 }
 
+void DivPlatformYM2203Ext::setCSM(bool isCSM) {
+  this->isCSM=isCSM;
+  if (isCSM) {
+    psgChanOffs++;
```

Comment:
Calling setCSM more than once will cause all sorts of bugs.
I know this doesn't happen in the code, but I still want to be sure.